Io box buttons 2

KorpelainenTommi opened the pull request:
Add new IOBoxButtons that are more like their design in Figma
-Fix a padding issue in ContentPanelActions

samikuikka committed:
login and register buttons to the navigation bar

samikuikka committed:
start of the login page, currently without validation and not coonnected to backend

samikuikka committed:
fixing input element

samikuikka committed:
moved cutominput and textarea to same folder

samikuikka committed:
Reusing prop information in textarea and custominput

samikuikka committed:
custom hoom for input and textarea for showing errors

samikuikka committed:
deleted unecessary import from textarea

samikuikka committed:
moved the PromptIOBox to ContentPanelPrompts component because it's not reusable in whole application

samikuikka committed:
custom hook for input elements + taking out the margin from filled button to fix bugs

samikuikka committed:
loggin error message v1

samikuikka committed:
closing the notification after a while from opening it

samikuikka committed:
small ui changes

Commit:
New IOBoxButtons

Commit:
Remove unrelated change, probably make this another pull request later if not lazy

KorpelainenTommi committed:
Remove errors from IOBoxBar since it's unused

KorpelainenTommi committed:
Merging main

KorpelainenTommi committed:
Prettier

KorpelainenTommi committed:
Prettier again, merge more stuff

Commit:
Merge branch 'pretty_print_structured_logging' of github.com:bytecraftoy/aalto-2022 into IOBoxButtons-2
Merge pretty_print_structured_logging

Commit:
Re-add changes that were lost when merging from main

itulkki commented on frontend/src/components/ContentPanel/ContentPanelPrompts/PromptIOBox/IOBoxBar/IOBoxButton/index.tsx:
```diff
@@ -25,18 +21,31 @@ export const IOBoxButton: React.FC<IOBoxButtonProps> = ({
+                name={nameVisible ? name : ''}
+                icon={icon}
+                colorPalette={colorPalette}
+                disabled={errors ? true : false}
```
Would
errors !== undefined //returns true for an empty string
or
Boolean(errors) //exactly same behaviour in this context
be prettier?

KorpelainenTommi replied:
Removed errors as it is not used by IOBoxButton. The button is not visible if there are errors, so there is no reason to disable it.

Commit:
Remove errors as it is redundant for IOBoxButtonm

KorpelainenTommi committed:
Added responsive design (#88)

* Added responsive design

* Merge bulk add (#89)

* Basic bulk adding functionality

Basic functionality for adding multiple boxes at once using pre-set buttons for adding 5 and 10 boxes at once.

* Dropdown menu

Added basic dropdown menu for secondary panel functions. Currently can add 5 or 10 I/O boxes.

* prettier on last commit

Prettier ran on previous commit

* Styling dropdown menu and cleaning the code

Changed the dropdown menu styling to utilize the color pallet and cleaned some redundant code.

* Styling updated

Further updated styling and removed the need for colorPalette in DropdownMenu

Co-authored-by: PyryManninen <pyry.manninen@gmail.com>
Co-authored-by: Tommi Korpelainen <tommi.korpelainen@aalto.fi>

Co-authored-by: Tommi Korpelainen <tommi.korpelainen@aalto.fi>
Co-authored-by: PyryManninen <pyry.manninen@gmail.com>
